src/commands/android/subcommands/disconnect/index.ts

```diff
@@ -0,0 +1,78 @@
+import colors from 'ansi-colors';
+import ADB from 'appium-adb';
+import inquirer from 'inquirer';
+
+import Logger from '../../../../logger';
+import {killEmulatorWithoutWait} from '../../adb';
+import {Options, Platform} from '../../interfaces';
+import {getBinaryLocation} from '../../utils/common';
+import {execBinarySync} from '../../utils/sdk';
+import {showConnectedRealDevices, showRunningAVDs} from '../common';
+
+export async function disconnect(options: Options, sdkRoot: string, platform: Platform) {
+  try {
+    const adbLocation = getBinaryLocation(sdkRoot, platform, 'adb', true);
+    const adb = await ADB.createADB({allowOfflineDevices: true});
+    const devices = await adb.getConnectedDevices();
+
+    if (devices.length === 0) {
+      Logger.log(`${colors.yellow('No device found running.')}`);
+
+      return true;
+    }
+
+    const devicesList = devices.map((device) => device.udid);
+
+    // Here, options.s represent the device id to disconnect.
+    // If the provided device id is not found then prompt the user to select the device.
+    if (options.s && typeof options.s === 'string') {
```

Comment:
We should use `deviceId` as the main flag name instead of `s`. `s` could be an alias to `deviceId`.